<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="deadzone_val"   default="0.1" />
    <arg name="device"   default="/dev/input/js0" />
    <arg name="topic_name" default="joy_default" />

    <arg name="button_map_name"   default="button_mapping_1.yaml" />
    <arg name="axis_map_name"   default="axis_mapping_3.yaml" />

    <node name="$(anon joystick)" pkg="joy" type="joy_node" output="screen" required="false">
        <param name="dev" value="$(arg device)" type="str"/>
        <param name="deadzone" value="0.0" />
        <param name="autorepeat_rate" value="30.0" />
        <remap from="joy" to="/$(arg topic_name)" />
        <remap from="joy/set_feedback" to="/$(arg topic_name)/set_feedback" />
    </node>
    <node name="$(anon joy_throttle)" pkg="topic_tools" type="throttle" required="false" args="messages /$(arg topic_name) 20 /$(arg topic_name)_throttled" />

    <group ns="bw" >
        <node name="$(anon bw_joystick)" pkg="bw_joystick" type="bw_joystick_node.py" output="screen" required="false">
            <param name="deadzone_joy_val" value="$(arg deadzone_val)" />
            <param name="joystick_topic" value="/$(arg topic_name)_throttled" />
            <param name="linear_scale" value="1.0" />
            <param name="angular_scale" value="5.0" />
            <param name="deadzone_joy_val" value="0.05" />

            <rosparam param="button_mapping" file="$(find bw_joystick)/config/$(arg button_map_name)" command="load"/>
            <rosparam param="axis_mapping" file="$(find bw_joystick)/config/$(arg axis_map_name)" command="load"/>

            <remap from="cmd_vel" to="cmd_vel_teleop"/>
        </node>
    </group>
</launch>
